datasetinsights/estimators/faster_rcnn.py

```diff
@@ -192,12 +197,23 @@ def train(self, train_data, val_data=None, **kwargs):
         val_loader = dataloader_creator(
             config, val_dataset, val_sampler, VAL, self.distributed
         )
-        self.train_loop(
-            train_dataloader=train_loader,
-            label_mappings=label_mappings,
-            val_dataloader=val_loader,
-            train_sampler=train_sampler,
-        )
+
+        params = self.config.clone()
+        if params.get(TrackerFactory.TRACKER):
+            params.pop(TrackerFactory.TRACKER)
+        self.mlflow_tracker.log_params(params)
+        try:
+            self.train_loop(
+                train_dataloader=train_loader,
+                label_mappings=label_mappings,
+                val_dataloader=val_loader,
+                train_sampler=train_sampler,
+            )
+        except Exception as e:
+
+            self.mlflow_tracker.end_run(status=TrackerFactory.RUN_FAILED)
+            raise e
+        self.mlflow_tracker.end_run()
```

Comment:
This might be better wrapped into the try/catch/except/finally clause to ensure it was always executed when `self.train_loop(...)` does not raise error.